### ISSUE TITLE
(0°C)^n should not equal 0*(°C)^n

### ISSUE THREAD

**Issue by ajkeller34** [ajkeller34/Unitful.jl#72]
Same for `°F`. Better to convert `°C` to an absolute scale like `K` before taking the power, if it is the only unit on the quantity.

Not sure if better to convert `°F` to `°Ra` or `K` since both are absolute scales, and `°Ra` is more closely related to `°F`...

**Comment by deszoeke** [comment]
Are you proposing that taking a power should always use an absolute temperature? Then the power operator would not have the expected relationship to multiplication for `Temperature`s.

This should also be done for dividing (the power is -1) or multiplying by temperature in many cases. In the ideal gas law, for example,
```
density=p/Rspecific*T^-(1//1), or
volume=Nmoles*Runiversal*T^(1//1)/p,
```
where T must be an absolute temperature. Yet this might not be universally desirable.

The proposed behavior is good for physics, but could lead to unexpected results, e.g. for evaluating polynomial fits linearized about a different temperature scale, like an empirical thermal response function of the form `r = a0+ a1*T + a2*T^2 + a3*T^3...` with T in °C.

Perhaps Unitful should always use absolute temperature upon (multiplication, division, or) taking a power and this empirical formula should be constructed from powers of `NoUnits` as 
`r = a0 + a1*NoUnits(T/u"°C") ...`. This would work as long as the ratio `T/u"°C" = u"°C"(T)/u"°C"` is not interpreted as absolute `u"K"(T)/u"°C"`, which this proposal seems dangerously close to doing.

**Comment by ajkeller34** [comment]
I would write your example as `r = ((1u"°C")/a_0) + ((T-T_0)/a_1)^1 + ((T-T_0)/a_2)^2 + ...`. In this way you can make an empirical fit around `T_0` where `T_0` is any temperature. By dividing by a coefficient before you take the power, the exponentiation would work properly. An added benefit is that all your coefficients now have the same units of temperature^1 and can be stored efficiently in an array. Finally, this expression will work regardless of the units of `T` for given `a_i`.

I am concerned that there may be cases where the wrong answer is given, including the one you point out, but ultimately I need Unitful to be useful for physics. It may be possible to make a custom `°C` unit with some associated logic to indicate whether it should be viewed as a temperature interval or as a temperature on a scale.

**Comment by deszoeke** [comment]
Something I like about Unitful is its descriptive explicit, rather than prescriptive implicit approach. Temperatures are problematic units. Some functions in my thermodynamics library require T in Celsius, some in Kelvin. Unitful handles this beautifully and consistently if the user is explicit about converting to the temperature units they expect. Otherwise Unitful calculates using the temperature scale in the unit of the quantity passed by default. Implicit automatic conversion to absolute units for some operations will lead to unexpected results.

Since an absolute temperature is expected, I prefer the more explicit `vbar = (k*u"K"(T)/mass)^(1//2)` to `vbar = (k*T/mass)^(1//2)`. Even with special handling of powers, Unitful would not know that the `^(1//2)` should convert `T` to Kelvin.

Absolute temperatures have their place, absolutely. For properties of water and things commonly measured on Earth, Celsius is more common and more precise. Using °C is just a standard special case of adding an offset to T, which I hope Unitful continues to handle gracefully. Specifying units with Unitful is an improvement on my code that had before added offsets.

**Comment by burnpanck** [comment]
See also #137: Absolute units (or units that *include* an *implicit* offset with respect to other units on the same scale) and relative units (units that *are* an offset) are fundamentally different things. For absolute units, many mathematical operations are ill-defined or have no physical meaning. That includes addition if multiple absolute units are involved or multiplication/exponentiation when there is even just one absolute unit involved. The special case of the physical laws involving absolute temperature are only well-defined because they in fact work on relative temperatures with respect to some specific implicit temperature origin, the absolute zero temperature (0 K). However, this additional information is not conveyed to a generic unit-system unless you write that offset explicitly as @ajkeller34 suggested, e.g. as `exp(-E/(k*(T-T0))`.
Due to this, I consider any units-library that *implicitly* inserts temperature offset adjustments as dangerously broken. Instead, either all temperatures should be considered relative temperatures, or there should be a dedicated type for absolute temperatures which disallows all ill-defined operations.

**Comment by RuiRojo** [comment]
Yeah. It's like times and dates. If physicists where in the habit of making formulas times, "time since the birth of the universe", and time since Christ was born, and treat them as the same thing but on different "units"

**Comment by ajkeller34** [comment]
Closed by #177. Unfortunately the solution will break some code I'm sure, but I do think it is an improvement, and in any case it is necessary in order to avoid misleading or unreliable results.
